pkg/cli/pod/pod.go

```diff
@@ -0,0 +1,414 @@
+			maxRestartLen = len(info.Restarts)
+		}
+		if len(info.CreationTimestamp) > maxAgeLen {
+			maxAgeLen = len(info.CreationTimestamp)
+		}
+	}
+	columnSpacing := 8
+	maxNameLen += columnSpacing
+	maxReadyLen += columnSpacing
+	maxStatusLen += columnSpacing
+	maxRestartLen += columnSpacing
+	maxAgeLen += columnSpacing
+	formatStr := fmt.Sprintf("%%-%ds%%-%ds%%-%ds%%-%ds%%-%ds\n", maxNameLen, maxReadyLen, maxStatusLen, maxRestartLen, maxAgeLen)
+	_, err := fmt.Fprintf(writer, formatStr, Name, Ready, Status, Restart, Age)
+	if err != nil {
+		fmt.Printf("Failed to print Pod information: %s.\n", err)
+		return
+	}
+	for _, info := range infoList {
+		_, err := fmt.Fprintf(writer, formatStr, info.Name, info.ReadyContainers, info.Status, info.Restarts, info.CreationTimestamp)
+		if err != nil {
+			fmt.Printf("Failed to print Pod information: %s.\n", err)
+			return
+		}
+	}
+}
+
+// createLabelSelector creates a label selector based on the provided job name or queue name.
+func createLabelSelector(jobName, queueName string) (labels.Selector, error) {
+	var labelSelector labels.Selector
+	reqs := []labels.Requirement{}
+
+	// Add label requirements based on jobName label
+	if jobName != "" {
+		inRequirement, err := labels.NewRequirement(v1alpha1.JobNameKey, selection.In, []string{jobName})
+		if err != nil {
+			return nil, err
+		}
+		reqs = append(reqs, *inRequirement)
+	}
+	// Add label requirements based on queueName label
+	if queueName != "" {
+		// pods under vcjobs
+		inRequirementForVcJob, err := labels.NewRequirement(v1alpha1.QueueNameKey, selection.In, []string{queueName})
+		if err != nil {
+			return nil, err
+		}
+		// pods under specific queue
+		inRequirementForPod, err := labels.NewRequirement(schedulingv1beta1.QueueNameAnnotationKey, selection.In, []string{queueName})
+		if err != nil {
+			return nil, err
+		}
+		reqs = append(reqs, *inRequirementForVcJob)
+		reqs = append(reqs, *inRequirementForPod)
+	}
+
+	if len(reqs) > 0 {
+		// If job name or queue name are provided, select pods with both labels.
+		labelSelector = labels.NewSelector().Add(reqs...)
+	} else {
+		// If job name is not provided, select all pods created by vcjobs.
+		inRequirement, err := labels.NewRequirement(v1alpha1.JobNameKey, selection.Exists, []string{})
+		if err != nil {
+			return nil, err
+		}
+		labelSelector = labels.NewSelector().Add(*inRequirement)
+	}
+	return labelSelector, nil
+}
+
+// translateTimestampSince translates a timestamp into a human-readable string using time.Since.
+func translateTimestampSince(timestamp metav1.Time) string {
+	if timestamp.IsZero() {
+		return "<unknown>"
+	}
+	return duration.HumanDuration(time.Since(timestamp.Time))
+}
+
+// PodInfo holds information about a pod.
+type PodInfo struct {
+	Name              string
+	ReadyContainers   string
+	Status            string
+	Restarts          string
+	CreationTimestamp string
+}
+
+// printPod information in a tabular format.
+func printPod(pod *corev1.Pod) PodInfo {
```

Comment:
If this part is copyed from k8s，please also comment to explain that.

Reply:
done